### DISCUSSION TITLE
How to get user input in Agent Group Chat

### DISCUSSION THREAD

**Discussion by deepinderdeol** [microsoft/semantic-kernel#9098] [Q&A] [answered]
Here's the example scenario:

1. Multiple agents start working together on user specified goal
2. Some work is completed but one of the agents does not have the information it needs to proceed
3. The user is notified about this and user input is requested
4. User provides the required information
5. Agent chat proceeds from where it left off at step 2 above

In an Office-Hours meeting I was told that the termination-criteria has to be used and that chat.IsCompleted should be set to false.

I will try the suggestion above, but an example showing a user-proxy-agent will be very helpful.

**Comment by evchaki** [comment, marked as answer]
@deepinderdeol - a sample in .NET is underway to show you how to do this.  You are on the right track with this --> chat.IsCompleted should be set to false

**Reply by deepinderdeol** [reply]
Thanks @evchaki

**Reply by deepinderdeol** [reply]
@evchaki checking back on this. Is the sample available somewhere? I could not find it.
Thanks.

**Comment by crickman** [comment]
Hi @deepinderdeol - We explore reseting the `IsCompleted` flag in this sample published on our Learn site:

https://learn.microsoft.com/en-us/semantic-kernel/frameworks/agent/examples/example-agent-collaboration?pivots=programming-language-csharp#the-chat-loop

**Reply by deepinderdeol** [reply]
Thanks @crickman - I will check it out.

**Reply by liquidgreetings** [reply]
Hey guys,

 I'm having a challenge with the sample referenced by @crickman.  I swapped out the builder.AddOpenAIChatCompletion(...) section in the sample with builder.AddOllamaChatCompletion(...) to test with an LLM on my local computer. But I'm getting an error.

The error happens in this code block: 
```
await foreach (ChatMessageContent response in chat.InvokeAsync())
{
    Console.WriteLine();
    Console.WriteLine($"{response.AuthorName.ToUpperInvariant()}:{Environment.NewLine}{response.Content}");
}
```
The error is:
```
System.MissingMethodException
  HResult=0x80131513
  Message=Method not found: '!!0 Microsoft.Extensions.AI.IChatClient.GetService(System.Object)'.
  Source=Microsoft.SemanticKernel.Connectors.Ollama
  StackTrace:
   at Microsoft.SemanticKernel.OllamaServiceCollectionExtensions.<>c__DisplayClass3_0.<AddOllamaChatCompletion>b__0(IServiceProvider serviceProvider, Object _)
   at Microsoft.Extensions.DependencyInjection.ServiceDescriptor.<>c__DisplayClass5_0.<.ctor>b__0(IServiceProvider sp)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteRuntimeResolver.VisitFactory(FactoryCallSite factoryCallSite, RuntimeResolverContext context)
   at M...
```

The following <ItemGroup> is from my project file. Note that it includes Microsoft.Extensions.AI.Ollama.I thought that including it in my project might resolve the error. However, when I added using Microsoft.Extensions.AI.Ollama; to my code, I got an error that reads: The type or namespace name 'Ollama' does not exist in the namespace 'Microsoft.Extensions.AI' (are you missing an assembly reference?).

   	<ItemGroup>
		<PackageReference Include="Azure.Identity" Version="1.13.1" />
		<PackageReference Include="Microsoft.Extensions.AI.Ollama" Version="9.0.1-preview.1.24570.5" />
		<PackageReference Include="Microsoft.Extensions.Configuration" Version="9.0.0" />
		<PackageReference Include="Microsoft.Extensions.Configuration.Binder" Version="9.0.0" />
		<PackageReference Include="Microsoft.Extensions.Configuration.EnvironmentVariables" Version="9.0.0" />
		<PackageReference Include="Microsoft.Extensions.Configuration.UserSecrets" Version="9.0.0" />
		<PackageReference Include="Microsoft.SemanticKernel.Agents.Core" Version="1.30.0-alpha" />
		<PackageReference Include="Microsoft.SemanticKernel.Connectors.AzureOpenAI" Version="1.30.0" />
		<PackageReference Include="Microsoft.SemanticKernel.Connectors.Ollama" Version="1.30.0-alpha" />
		<PackageReference Include="Microsoft.VisualStudio.Azure.Containers.Tools.Targets" Version="1.21.0" />
	</ItemGroup>

The sample really looks good and is pretty much exactly what I'm looking to accomplish. Is there a C# agent collaboration sample that uses Ollama Chat Completion instead of OpenAI? Or is there a simple fix for this error? Any help in resolving this issue will be greatly appreciated. Thanks in advance.
